### PR TITLE
[OYPD-202] Theme updates for the location finder, adds templates, update styling…

### PR DESCRIPTION
**Pull request by david-hernandez**
- [x] Make sure your site has branch, camp, and facility content nodes.
- [x] Create landing page.
- [x] Add location finder filters paragraph to the header area.
- [x] Add location finder paragraph to the content area.
- [x] View page.

screen shot 2017-02-14 at 10 18 36 pm

screen shot 2017-02-14 at 10 21 27 pm

screen shot 2017-02-14 at 10 21 34 pm

screen shot 2017-02-14 at 10 25 20 pm

We may need to discuss the buttons. They are blue because of the color module implementation changes. I can add the btn-default styling to the colors.css file, but I'm concerned about any regressions that might cause.

### CODE REVIEW

**Comment by gundevel** [comment]
Build comment file: 
--------------
===============
<strong>2</strong>  errors for CodeSniffer: Drupal standard file http://openy-dev.ffwua.com/build469/build_reports/Drupalsniff.txt
Build site installed at http://openy-dev.ffwua.com/build469

--------------
Test PASSed.
Refer to this link for build results (access rights to CI server needed): 
http://openy-dev.ffwua.com:8080/job/PR_BUILDER/469/

**Comment by gundevel** [comment]
Behat test results file URL: http://openy-dev.ffwua.com/build469/profiles/openy/build/reports/behat

**Comment by alexschedrov** [comment]
1. CodeSniffer notices
2. Bundle buttons should be have gray color
image

3. When "Coming soon" is selected, teaser should have following design + Branch title should lead to temporary URL.
image

Other than that looks awesome!

**Comment by gundevel** [comment]
Build comment file: 
--------------
===============
Build site installed at http://openy-dev.ffwua.com/build473

--------------
Test PASSed.
Refer to this link for build results (access rights to CI server needed): 
http://openy-dev.ffwua.com:8080/job/PR_BUILDER/473/

**Comment by gundevel** [comment]
Behat test results file URL: http://openy-dev.ffwua.com/build473/profiles/openy/build/reports/behat

**Comment by david-hernandez** [comment]
The last commit should handle the codesniffer error and the coming soon.

screen shot 2017-02-15 at 9 23 33 am

**Comment by david-hernandez** [comment]
Last commit adds white buttons and temp url. There is a problem where the buttons are going to get overridden if the main background color is changed. I haven't figure out how to separate them. They both start out with the same color.

Screenshot show I added a temp url that appears when I hovered over the title.

screen shot 2017-02-15 at 5 03 13 pm

**Comment by gundevel** [comment]
Build comment file: 
--------------
===============
Build site installed at http://openy-dev.ffwua.com/build477

--------------
Test PASSed.
Refer to this link for build results (access rights to CI server needed): 
http://openy-dev.ffwua.com:8080/job/PR_BUILDER/477/

**Comment by gundevel** [comment]
Behat test results file URL: http://openy-dev.ffwua.com/build477/profiles/openy/build/reports/behat

**Comment by Sanchiz** [comment]
retest this please

**Comment by bjwilkins** [comment]
Build comment file: 
--------------
===============
Build site installed at http://openy-dev.ffwua.com/build479

--------------
Test PASSed.
Refer to this link for build results (access rights to CI server needed): 
http://openy-dev.ffwua.com:8080/job/PR_BUILDER/479/

File: themes/openy_themes/openy_rose/templates/node/node--branch--teaser.html.twig [hunk 1]
```diff
@@ -81,13 +81,12 @@
...
 %}
 <article{{ attributes.addClass(classes) }}>
-
   {{ content.field_location_area }}
 
   {{ title_prefix }}
   {% if not page %}
     <h2{{ title_attributes }}>
-      <a href="{{ url }}" rel="bookmark">{{ label }}</a>
+      <a href="{{ node.field_location_temp_url.0.url ? node.field_location_temp_url.0.url : url }}" rel="bookmark">{{ label }}</a>
```
Comment (alexschedrov) [inline themes/openy_themes/openy_rose/templates/node/node--branch--teaser.html.twig]: 1. This should be applied only when branch is "Coming Soon".
2. Let's set variable using `set` and the re-use it. `node.field_location_temp_url.0.url` looks a little bit scary.

**Review by alexschedrov — CHANGES_REQUESTED** [review]

**Comment by bjwilkins** [comment]
Behat test results file URL: http://openy-dev.ffwua.com/build479/profiles/openy/build/reports/behat

File: themes/openy_themes/openy_rose/templates/field/field--node--field-location-state.html.twig [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+  {% if multiple %}
+    <div{{ attributes }}>
+      {% for item in items %}
+        <div{{ item.attributes }}>{{ item.content }}</div>
+      {% endfor %}
+    </div>
+  {% else %}
+    {% for item in items %}
+      {% if item.content['#markup']|length %}
+        <div{{ attributes.addClass(field_name) }}><i class="fa fa-star"></i><span>New Y</span>{{ item.content }}</div>
```
Comment (alexschedrov) [inline themes/openy_themes/openy_rose/templates/field/field--node--field-location-state.html.twig]: `New Y` should be translatable.

**Review by alexschedrov — CHANGES_REQUESTED** [review]

**Comment by bjwilkins** [comment]
Build comment file: 
--------------
===============
Build site installed at http://openy-dev.ffwua.com/build482

--------------
Test PASSed.
Refer to this link for build results (access rights to CI server needed): 
http://openy-dev.ffwua.com:8080/job/PR_BUILDER/482/

**Comment by david-hernandez** [comment]
Added translation for coming soon text. Added logic to only use the temporary url when coming soon is activated for the node.

The `node.field_location_temp_url.0.url` is unfortunately needed. The url field does not supply the url separate from the text. There isn't a widget option for that. Also, the other cleaner options like `field_location_temp_url.uri` or `url` can return either an external url or a route. There is a url() function to deal with a route but not an external url. So every method I tried resulted in having to just dig into the url array.

**Comment by bjwilkins** [comment]
Behat test results file URL: http://openy-dev.ffwua.com/build482/profiles/openy/build/reports/behat

**Comment by AlexNetman** [comment]
Please, remove the scroll from pins
image

**Comment by alexschedrov** [comment]
image

**Comment by AlexNetman** [comment]
image

Regarding "Coming soon" link -  I think it's better to just make redirect from node. But not updating all those view modes. Global redirect from location node to that URL would work for all other places were link to location might appear - menus, other landing pages, blogs associated with the branch.

**Comment by bjwilkins** [comment]
Build comment file: 
--------------
===============
Build site installed at http://openy-dev.ffwua.com/build484

--------------
Test PASSed.
Refer to this link for build results (access rights to CI server needed): 
http://openy-dev.ffwua.com:8080/job/PR_BUILDER/484/

**Comment by bjwilkins** [comment]
Behat test results file URL: http://openy-dev.ffwua.com/build484/profiles/openy/build/reports/behat

**Review by alexschedrov — APPROVED** [review]
Approved.

**Pull request merged**